### DISCUSSION TITLE
How to handle the space in secret value in github settings

### DISCUSSION THREAD

**Discussion by Basha963** [community/community#27113] [Actions]
<p>How to handle the space in secret value in github settings,<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/f/7f0b24abac9f10032f5a922b91d0195f5cf0b0e7.png" data-download-href="/uploads/short-url/i7SvO7p60zDHp1lvDFS4Lyr27b1.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/66317892/181101756-3a6350bd-b0d1-4840-a19e-a64fe1bd5c91.png" alt="image" data-base62-sha1="i7SvO7p60zDHp1lvDFS4Lyr27b1" width="690" height="327" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/f/7f0b24abac9f10032f5a922b91d0195f5cf0b0e7_2_690x327.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/f/7f0b24abac9f10032f5a922b91d0195f5cf0b0e7_2_1035x490.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/f/7f0b24abac9f10032f5a922b91d0195f5cf0b0e7_2_1380x654.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/f/7f0b24abac9f10032f5a922b91d0195f5cf0b0e7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1882×892 27.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am getting the below error, by using secret token above format. Getting the issue with space between secret value “Token xxxxxxxxxxx”. Please help me how to handle this scenario. Thank!</p>
<p>Unable to find image ‘xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx:latest’ locally</p>
<p><a href="https://github.com/xxxxx/xxxxxxx/runs/xxxxxxxxxx?check_suite_focus=true#step:15:43" rel="noopener nofollow ugc">43</a>docker: Error response from daemon: pull access denied for xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx, repository does not exist or may require ‘docker login’: denied: requested access to the resource is denied.</p>

**Comment by airtower-luna** [comment]
<p>How are you trying to use that secret? Usually the secret value should only contain the part that’s supposed to be secret, not a prefix like “Token”.</p>

**Comment by Basha963** [comment]
<p>actually one of the API we are using Authorization value like “Token 7555xxxxxxxx”. I added this value in github setting secrets. And using workflow yml file to load this secret values.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="Basha963" data-post="3" data-topic="182336">
<div class="title">
<div class="quote-controls"></div>
 Basha963:</div>
<blockquote>
<p>actually one of the API we are using Authorization value like “Token 7555xxxxxxxx”.</p>
</blockquote>
</aside>
<p>Do you mean an Authorization HTTP header?</p>
<aside class="quote no-group" data-username="Basha963" data-post="3" data-topic="182336">
<div class="title">
<div class="quote-controls"></div>
 Basha963:</div>
<blockquote>
<p>And using workflow yml file to load this secret values.</p>
</blockquote>
</aside>
<p>Yes, but exactly how? I assume there’s some step that involves something like</p>
<pre><code class="lang-auto">${{ secrets.MY_TOKEN }}
</code></pre>
<p>somewhere, and the details of that are important here.</p>

**Comment by Basha963** [comment]
<p>Yes, it’s authorization HTTP header and I am using secret like below</p>
<ul>
<li>name: Load secrets<br>
run: |</li>
</ul>
<p>echo “TEST_PROD_TOKEN =${{ TEST_PROD_TOKEN }}” &gt;&gt; $GITHUB_ENV</p>
<p>docker run --rm -v ${PWD}:/app <br>
–env TEST_PROD_TOKEN=${TEST_PROD_TOKEN} \</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Okay, in that case I see three problems here:</p>
<ol>
<li>Having “Token” as part of the secret. That can cause trouble with masking secrets in the logs, and the documentation warns against it.</li>
<li>You can’t use an environment value written to <code>GITHUB_ENV</code> in the same step that sets it, only steps after.</li>
<li>The <code>docker run</code> command contains an unquoted space.</li>
</ol>
<p>So, what you should do:</p>
<ol>
<li>Remove the “Token” part from your secret. The value should only be the “7555xxxxxxxx” part.</li>
<li>You can then use it in your workflow like this (assuming the secret is also called <code>TEST_PROD_TOKEN</code>) without the detour via <code>GITHUB_ENV</code>:</li>
</ol>
<pre><code class="lang-yaml">- name: run docker with token
  run: |
    docker run --rm -v ${PWD}:/app --env="TEST_PROD_TOKEN=Token ${{ secrets.TEST_PROD_TOKEN}}" ...
</code></pre>
<p>Note the quotes around the value for <code>--env</code>, they keep the value from getting cut off at the space after “Token”.</p>

**Comment by Basha963** [comment]
<p>Thank you very much <a class="mention" href="/airtower-luna">@airtower-luna</a>  for your help. Now I am able to run all my tests without issues.</p>
